### PR TITLE
Refactor split into ranges

### PR DESCRIPTION
**Pull request by nkohen**

### CODE REVIEW

**Review by benthecarman — APPROVED** [review]

File: core-test/src/test/scala/org/bitcoins/core/protocol/dlc/CETCalculatorTest.scala [hunk 1]
```diff
@@ -33,15 +33,13 @@ class CETCalculatorTest extends BitcoinSUnitTest {
...
       VariablePayoutRange(21, 39),
       ConstantPayoutRange(40, 50),
-      VariablePayoutRange(51, 69),
-      ZeroPayoutRange(70, 70),
-      VariablePayoutRange(71, 79),
+      VariablePayoutRange(51, 79),
       ConstantPayoutRange(80, 90),
       VariablePayoutRange(91, 98),
-      MaxPayoutRange(99, 108),
+      ConstantPayoutRange(99, 108), // totalCollateral
```
Comment (Christewart) [inline core-test/src/test/scala/org/bitcoins/core/protocol/dlc/CETCalculatorTest.scala:42]: nit: Maybe we should just add a `payout: Satoshis` field to `ConstantPayoutRange()`?

File: core/src/main/scala/org/bitcoins/core/protocol/dlc/CETCalculator.scala [hunk 2]
```diff
@@ -66,21 +57,10 @@ object CETCalculator {
...
-      if (value <= Satoshis.zero) {
-        ZeroPayoutRange(index, index)
-      } else if (value >= totalCollateral) {
-        MaxPayoutRange(index, index)
-      } else {
-        VariablePayoutRange(index, index)
-      }
-    }
+  /** A CETRange with a single element range */
+  case class SingletonPayoutRange(index: Long) extends CETRange {
```
Comment (Christewart) [inline core/src/main/scala/org/bitcoins/core/protocol/dlc/CETCalculator.scala:61]: :+1:

**Review by Christewart — COMMENTED** [review]
My code review: 

https://github.com/Christewart/bitcoin-s-core/tree/2021-02-04-breakout-processconstant

basically 

1. Move `processConstant()` out of `splitIntoRanges()`
2. Add a type for what is returned from `processConstant()`.

File: core-test/src/test/scala/org/bitcoins/core/protocol/dlc/CETCalculatorTest.scala (hunk 1 above)
Comment (nkohen) [reply core-test/src/test/scala/org/bitcoins/core/protocol/dlc/CETCalculatorTest.scala:42]: I don't see why that would be needed? This comment isn't important or anything

**Comment by nkohen** [comment]
Responded @Christewart

File: core-test/src/test/scala/org/bitcoins/core/protocol/dlc/CETCalculatorTest.scala (hunk 1 above)
Comment (Christewart) [reply core-test/src/test/scala/org/bitcoins/core/protocol/dlc/CETCalculatorTest.scala:42]: it's nice to know what the payout was for a specific range. Seems like minimal storage, where the alternative is having to get re-evaluate the function again, but :man_shrugging:

**Review by Christewart — APPROVED** [review]

**Pull request merged**